File: pyHeirarchy/jewelry.py
class Jewelry:
    def __init__ (self, polished : bool = True):
        self._polished : bool = polished
    @property
    def polished(self) -> bool:
        return self._polished

    def polish(self) -> None:
        self._polished = True

    def wear(self) -> None:
        self._polished = False

    

File: cppHeirarchy/src/jewelry.cpp
#include "jewelry.h"


namespace bling {

    Jewelry::Jewelry(bool _polished) : polished(_polished){

    }

    void Jewelry::polish(){
        polished = true;
    }
    void Jewelry::wear(){
        polished = false;
    }

    bool Jewelry::isPolished() const {
        return polished;
    }

}